### PR TITLE
Move topology_test to feature dir & update READMEs

### PR DESCRIPTION
**Pull request by greg-dennis**
This fixes https://github.com/openconfig/featureprofiles/issues/1956

### CODE REVIEW

**Comment by bstoll** [comment]
The virtual tests are failing with this error:

```
topology_test.go:193: did not get expected ATE telemetry, got: [UP UP UP UP], want: [UP UP]
```

should this be using TESTBED_DUT_ATE_2LINKS in metadata.textproto?

**Comment by greg-dennis** [comment]
> The virtual tests are failing with this error:
> 
> ```
> topology_test.go:193: did not get expected ATE telemetry, got: [UP UP UP UP], want: [UP UP]
> ```
> 
> should this be using TESTBED_DUT_ATE_2LINKS in metadata.textproto?

I just saw! Great that the KNE tests caught a problem with the test! The test is otherwise written to be independent of the number of ports, so I think the better fix is to change the check.

**Comment by greg-dennis** [comment]
Done. Made the test independent of the number of ports and cleaned up various things in the logic, too.

File: feature/example/tests/topology_test/topology_test.go [hunk 1]
```diff
@@ -188,9 +181,11 @@ func TestTopology(t *testing.T) {
 
 	// Query ATE telemetry.
 	t.Run("ATE Telemetry", func(t *testing.T) {
-		want := []otgoc.E_Port_Link{otgoc.Port_Link_UP, otgoc.Port_Link_UP}
-		if got := gnmi.GetAll(t, ate.OTG(), gnmi.OTG().PortAny().Link().State()); !cmp.Equal(got, want) {
-			t.Errorf("did not get expected ATE telemetry, got: %v, want: %v", got, want)
+		const want = otgoc.Port_Link_UP
+		for _, ap := range ate.Ports() {
+			if got := gnmi.Get(t, ate.OTG(), gnmi.OTG().Port(ap.Name()).Link().State()); got != want {
```
Comment (bstoll) [inline feature/example/tests/topology_test/topology_test.go:186]:
```go
if got := gnmi.Get(t, ate.OTG(), gnmi.OTG().Port(ap.ID()).Link().State()); got != want {
```

works for me.  `ap.Name()` does not.

**Review by bstoll — APPROVED** [review]

**Pull request merged**